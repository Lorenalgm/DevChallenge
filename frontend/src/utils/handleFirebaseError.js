export const AuthErrorHandler = {
    generic:
        'Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.',
    'auth/weak-password':
        'Senha digitada é muito fraca. Tente usar mais de 6 caracteres com números e caracteres especiais.',
    'auth/email-already-in-use': 'O email fornecido ja está em uso.',
    'auth/email-already-exists':
        'O e-mail fornecido já está em uso por outro usuário.',
    'auth/internal-error':
        'Erro interno do servidor. Tente novamente mais tarde.',
    'auth/invalid-email': 'Email inválido.',
    'auth/user-not-found':
        'Não há registro de usuário existente correspondente ao identificador fornecido.',
    'auth/wrong-password': 'Senha digitada não esta correta',
    'resource-exhausted':
        'Servidores em manutenção. Se esse erro persistir, contate o administrador do sistema.',
};
