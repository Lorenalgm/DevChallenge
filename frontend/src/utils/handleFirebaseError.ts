export const AuthErrorHandler = (option: string) => {
  switch (option) {
    case "generic":
      return "Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.";
    case "auth/weak-password":
      return "Senha digitada é muito fraca. Tente usar mais de 6 caracteres com números e caracteres especiais.";
    case "auth/email-already-in-use":
      return "O email fornecido ja está em uso.";
    case "auth/email-already-exists":
      return "O e-mail fornecido já está em uso por outro usuário.";
    case "auth/internal-error":
      return "Erro interno do servidor. Tente novamente mais tarde.";
    case "auth/invalid-email":
      return "Email inválido.";
    case "auth/user-not-found":
      return "Não há registro de usuário existente correspondente ao identificador fornecido.";
    case "auth/wrong-password":
      return "Senha digitada não esta correta";
    case "resource-exhausted":
      return "Servidores em manutenção. Se esse erro persistir, contate o administrador do sistema.";
    default:
      return "Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.";
  }
};
