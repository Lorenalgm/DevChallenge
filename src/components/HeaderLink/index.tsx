import Link from 'next/link'
import { ListItem, Link as ChakraLink, LinkProps } from '@chakra-ui/react'

export type HeaderLinkProps = {
  active: boolean
  href: string
} & LinkProps

const HeaderLink = ({
  active,
  href,
  children,
  mt,
  ...props
}: HeaderLinkProps) => (
  <ListItem mt={mt}>
    <Link href={href} passHref>
      <ChakraLink
        color={active ? 'primary.400' : 'white'}
        fontSize="xl"
        textDecoration="none !important"
        _hover={{ opacity: '0.8' }}
        {...props}
      >
        {children}
      </ChakraLink>
    </Link>
  </ListItem>
)

export default HeaderLink
