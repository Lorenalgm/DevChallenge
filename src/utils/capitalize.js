export default function capitalize(word) {
    if (!word) return undefined;

    const firstLetter = word[0];
    const restOfWord = word.slice(1);

    return firstLetter.toUpperCase() + restOfWord;
}
