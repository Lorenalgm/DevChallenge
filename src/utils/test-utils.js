import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import { render } from '@testing-library/react';

import { GlobalStyles } from '../styles/GlobalStyles';

const customRender = (component, { ...renderOptions } = {}) =>
    render(
        <>
            <GlobalStyles />
            {component}
            <ToastContainer />
        </>,
        { wrapper: MemoryRouter, ...renderOptions }
    );

export * from '@testing-library/react';
export { customRender as render };
