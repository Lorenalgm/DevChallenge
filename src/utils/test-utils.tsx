import { render, RenderResult } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'
import { matchers } from '@emotion/jest'

expect.extend(matchers)

import theme from 'theme'

export const customRender = (children: React.ReactNode): RenderResult =>
  render(
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  )

export * from '@testing-library/react'
export { customRender as render }
