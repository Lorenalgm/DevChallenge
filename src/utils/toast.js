import { toast } from 'react-toastify';

const messages = {
  error(msg) {
    return msg || 'Ocorreu um erro no processamento';
  },
  success(msg) {
    return msg || 'Sucesso';
  },
};

const ToastNotification = {
  notify: (method, msg) => {
    const methodStr = method;
    method = toast[method];
    method(messages[methodStr](msg));
  },
};

export default ToastNotification;
