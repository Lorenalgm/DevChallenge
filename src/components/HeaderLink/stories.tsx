import { Story, Meta } from '@storybook/react/types-6-0'
import { List } from '@chakra-ui/react'

import HeaderLink, { HeaderLinkProps } from '.'

export default {
  title: 'HeaderLink',
  component: HeaderLink
} as Meta

export const Default: Story<HeaderLinkProps> = (args) => (
  <List>
    <HeaderLink {...args} />
  </List>
)

Default.args = {
  href: '/',
  children: 'HeaderLink',
  active: true
}
