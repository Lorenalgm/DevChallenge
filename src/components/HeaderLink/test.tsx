import { render, screen } from 'utils/test-utils'
import { List } from '@chakra-ui/react'

import HeaderLink from '.'

describe('<HeaderLink />', () => {
  it('should render active link', () => {
    const { container } = render(
      <List>
        <HeaderLink active href="/">
          my link
        </HeaderLink>
      </List>
    )

    expect(screen.getByRole('link', { name: /my link/i })).toHaveAttribute(
      'href',
      '/'
    )

    expect(container.firstChild?.firstChild).toMatchSnapshot()
  })

  it('should render not active link', () => {
    render(
      <List>
        <HeaderLink active={false} href="/my-link">
          link
        </HeaderLink>
      </List>
    )

    expect(screen.getByRole('link', { name: /link/i })).toHaveAttribute(
      'href',
      '/my-link'
    )
  })
})
