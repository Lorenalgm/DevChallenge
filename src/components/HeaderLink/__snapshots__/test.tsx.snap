// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLink /> should render active link 1`] = `
.emotion-1 {
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  cursor: pointer;
  -webkit-text-decoration: none!important;
  text-decoration: none!important;
  outline: 2px solid transparent;
  outline-offset: 2px;
  color: var(--chakra-colors-primary-400);
  font-size: var(--chakra-fontSizes-xl);
}

.emotion-1:hover,
.emotion-1[data-hover] {
  opacity: 0.8;
}

.emotion-1:focus,
.emotion-1[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

<li
  class="emotion-0"
>
  <a
    class="chakra-link emotion-1"
    href="/"
  >
    my link
  </a>
</li>
`;
